onnx/shape_inference/implementation.cc

```diff
@@ -175,13 +251,14 @@ static void InferShapesImpl(
     }
   }
 
-  std::unordered_map<std::string, const TensorProto*> inputDataByName;
   // save for free memory
-  std::vector<TypeProto*> initializerTypeList;
+  std::vector<std::unique_ptr<TypeProto>> initializerTypeList;
```

Comment:
do we even need a unique_ptr here? Can the vector simply hold the TypeProto object?